parsl/monitoring/db_manager.py

```diff
@@ -301,78 +311,83 @@ def start(self, priority_queue, node_queue, resource_queue):
-            if messages:
+            node_info_messages = self._get_messages_in_batch(self.pending_node_queue,
+                                                             interval=self.batching_interval,
+                                                             threshold=self.batching_threshold)
+            if node_info_messages:
                 logger.debug(
-                    "Got {} messages from node queue".format(len(messages)))
-                self._insert(table=NODE, messages=messages)
+                    "Got {} messages from node queue".format(len(node_info_messages)))
+                self._insert(table=NODE, messages=node_info_messages)
 
             """
             Resource info messages
 
             """
-            messages = self._get_messages_in_batch(self.pending_resource_queue,
-                                                   interval=self.batching_interval,
-                                                   threshold=self.batching_threshold)
+            resource_messages = self._get_messages_in_batch(self.pending_resource_queue,
+                                                            interval=self.batching_interval,
+                                                            threshold=self.batching_threshold)
 
-            if messages or first_messages:
+            if resource_messages:
                 logger.debug(
-                    "Got {} messages from resource queue".format(len(messages)))
-                self._insert(table=RESOURCE, messages=messages)
-                for msg in messages:
+                    "Got {} messages from resource queue, {} reprocessable".format(len(resource_messages), len(reprocessable_first_resource_messages)))
+                self._insert(table=RESOURCE, messages=resource_messages)
+                for msg in resource_messages:
                     if msg['first_msg']:
+
                         msg['task_status_name'] = States.running.name
                         msg['task_time_running'] = msg['timestamp']
+
                         if msg['task_id'] in inserted_tasks:
-                            first_messages.append(msg)
+                            reprocessable_first_resource_messages.append(msg)
                         else:
-                            left_messages[msg['task_id']] = msg
-                if first_messages:
-                    self._insert(table=STATUS, messages=first_messages)
-                    self._update(table=TASK,
-                                 columns=['task_time_running',
-                                          'run_id', 'task_id',
-                                          'hostname'],
-                                 messages=first_messages)
+                            if msg['task_id'] in deferred_resource_messages:
+                                logger.error("Task {} already has a deferred resource message. Discarding previous message.".format(msg['task_id']))
```

Comment:
This is a good point. Do you see this quite often? I assume it would happen when the `resource_monitoring_interval` is very small. 

I wondered if it is appropriate to discard any of these messages. But anyway, this is not related to this PR.

Reply:
I have never seen it in real life but I wanted to log that this happened rather than silently discarding.